cfn/encoding/unstringify.go

```diff
@@ -3,9 +3,151 @@ package encoding
 import (
 	"fmt"
 	"reflect"
+	"strconv"
 	"strings"
 )
 
+func convertStruct(i interface{}, t reflect.Type, pointer bool) (reflect.Value, error) {
+	m, ok := i.(map[string]interface{})
+	if !ok {
+		return zeroValue, fmt.Errorf("Cannot convert %T to struct", i)
+	}
+
+	out := reflect.New(t)
+
+	Unstringify(m, out.Interface())
+
+	if !pointer {
+		out = out.Elem()
+	}
+
+	return out, nil
+}
+
+func convertString(i interface{}, pointer bool) (reflect.Value, error) {
+	s, ok := i.(string)
+
+	if !ok {
+		return zeroValue, fmt.Errorf("Cannot convert %T to string", i)
+	}
+
+	if pointer {
+		return reflect.ValueOf(&s), nil
+	}
+
+	return reflect.ValueOf(s), nil
+}
+
+func convertBool(i interface{}, pointer bool) (reflect.Value, error) {
+	var b bool
+	var err error
+
+	switch v := i.(type) {
+	case bool:
+		b = v
+
+	case string:
+		b, err = strconv.ParseBool(v)
+		if err != nil {
+			return zeroValue, err
+		}
+
+	default:
+		return zeroValue, fmt.Errorf("Cannot convert %T to bool", i)
+	}
+
+	if pointer {
+		return reflect.ValueOf(&b), nil
+	}
+
+	return reflect.ValueOf(b), nil
+}
+
+func convertInt(i interface{}, pointer bool) (reflect.Value, error) {
+	var n int
+
+	switch v := i.(type) {
+	case int:
+		n = v
+
+	case float64:
+		n = int(v)
+
+	case string:
+		n64, err := strconv.ParseInt(v, 0, 32)
+		if err != nil {
+			return zeroValue, err
+		}
+
+		n = int(n64)
+
+	default:
+		return zeroValue, fmt.Errorf("Cannot convert %T to bool", i)
+	}
+
+	if pointer {
+		return reflect.ValueOf(&n), nil
+	}
+
+	return reflect.ValueOf(n), nil
+}
+
+func convertFloat64(i interface{}, pointer bool) (reflect.Value, error) {
+	var f float64
+	var err error
+
+	switch v := i.(type) {
+	case float64:
+		f = v
+
+	case int:
+		f = float64(v)
+
+	case string:
+		f, err = strconv.ParseFloat(v, 64)
+		if err != nil {
+			return zeroValue, err
+		}
+
+	default:
+		return zeroValue, fmt.Errorf("Cannot convert %T to bool", i)
+	}
+
+	if pointer {
+		return reflect.ValueOf(&f), nil
+	}
+
+	return reflect.ValueOf(f), nil
+}
+
+func convertType(t reflect.Type, i interface{}) (reflect.Value, error) {
```

Comment:
Can we also include the array type? The plug support the conversion the in the code generator.

Reply:
Just put that in :)